src/pages/product/index.jsx

```diff
@@ -1,20 +1,52 @@
 import React from 'react';
-import { useLocation } from 'react-router-dom';
-import { Container, Image, InformationContainer, Text } from './styles';
-
+import { useParams } from 'react-router-dom';
+import { useProductByIdQuery } from '../../api/products/getProductByID.js';
+import SpinnerLoading from '../../components/SpinnerLoading';
+import { PRODUCT_NOT_FOUND } from '../../global/strings';
+import {
+  Container,
+  Image,
+  InformationContainer,
+  Name,
+  Description,
+  Price,
+  Category,
+  Size,
+  AddToCartButton,
+  NotFoundContainer,
+  NotFoundMessage,
+} from './styles';
 export default function Product() {
-  const location = useLocation();
-  const product = location.state;
+
+  const { id } = useParams();
+  const { product, isLoading, error, isErrorProductNotFound } =
+    useProductByIdQuery(id);
 
   return (
-    <Container>
-      <Image src={product.image} />
-      <InformationContainer>
-        <Text>Price : {product.price}</Text>
-        <Text>Description : {product.description}</Text>
-        <Text>Category : {product.category}</Text>
-        <Text>Size : {product.size}</Text>
-      </InformationContainer>
-    </Container>
+    <>
+      {isLoading ? (
+        <SpinnerLoading top="50%" right="50%" />
+      ) : (
+        <Container>
+          {isErrorProductNotFound || error ? (
+            <NotFoundContainer>
+              <NotFoundMessage>{PRODUCT_NOT_FOUND}</NotFoundMessage>
+            </NotFoundContainer>
+          ) : (
+            <>
+              <Image src={product.data.image} />
+              <InformationContainer>
```

Comment:
lets make this a seoparate component calle it InformationContainer and pass product as props then inside this component :
. destructe all properties instead of accessing it as product.data.price
. add functions inside string file for dispalying each property ex:
    const DISPLAY_PRICE=(price)=>`Price: ${price} $`